tools/resourcedocsgen/pkg/util/language/language.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package language exposes an enumeration of docs generation target languages:
+//
+// - [CSharp]
+// - [Go]
+// - [Java]
+// - [Python]
+// - [Typescript]
+// - [YAML]
+//
+// For a list of all languages, use [ListAll].
+package language
+
+// Language is a valid docsgen language. The default value is not valid.
+type Language struct{ tag string }
+
+var (
+	CSharp     = Language{"csharp"}
+	Go         = Language{"go"}
+	Java       = Language{"java"}
+	Python     = Language{"python"}
+	Typescript = Language{"nodejs"}
+	YAML       = Language{"yaml"}
+)
+
+// Iteration order of a [Set] of languages is well defined.
+//
+// When-ever a new language is added above, it needs to be added here.
+var langOrder = [...]Language{
+	CSharp,
+	Go,
+	Typescript,
+	Python,
+	YAML,
+	Java,
+}
+
+func (l Language) String() string {
+	if l.tag == "" {
+		return "invalid"
+	}
+	return l.tag
+}
+
+func (l Language) GoString() string {
+	var varName string
+	switch l {
+	case CSharp:
+		varName = "CSharp"
+	case Go:
+		varName = "Go"
+	case Java:
+		varName = "Java"
+	case Python:
+		varName = "Python"
+	case Typescript:
+		varName = "Typescript"
+	case YAML:
+		varName = "YAML"
+	default:
+		varName = "Language{}"
+	}
+
+	return "language." + varName
+}
```

Comment:
Is this used anywhere? Maybe GitHub is hiding a diff from me but I can't see it.

Reply:
`GoString` is the function invoked by the `%#v` symbol with `fmt.*` functions. It's used in some `panic`s, but never invoked during normal runtime. It's nice to have when debugging.

It allows `fmt.Printf("Language: %#v\n", language.YAML)` to print `language.YAML` instead of `language.Language{tag: "yaml"}`.